graylog2-web-interface/src/views/components/datatable/DataTable.tsx

```diff
@@ -128,6 +163,32 @@ const DataTable = ({
     return Promise.reject();
   }, [config, widget, editing, formContext]);
 
+  const togglePin = useCallback((field: string) => {
+    const dirty = formContext?.dirty;
+
+    const updateWidget = () => {
+      const curVisualizationConfig = widget.config.visualizationConfig || DataTableVisualizationConfig.create([]).toBuilder().build();
+      let pinned_columns;
+
+      if (curVisualizationConfig?.pinned_columns?.has(field)) {
+        pinned_columns = curVisualizationConfig.pinned_columns.delete(field);
+      } else {
+        pinned_columns = curVisualizationConfig.pinned_columns.add(field);
+      }
+
+      return WidgetActions.updateConfig(widget.id, widget.config.toBuilder().visualizationConfig(curVisualizationConfig.toBuilder().pinned_columns(pinned_columns.toJS()).build()).build());
+    };
+
+    if (!editing || (editing && !dirty)) return updateWidget();
```

Comment:
Please do not omit curly braces for single-line if-bodies. They are very helpful for understanding the structure of the code visually IMO.